src/app/lib/business/bugsee/bugsee_manager.dart

```diff
@@ -0,0 +1,210 @@
+
+  /// indicate if bugsee configuration is valid
+  /// config is valid if app in release mode and the provided token is valid
+  /// following the [bugseeTokenFormat] regex.
+  bool get isConfigurationValid;
+
+  /// initialize bugsee with given token
+  /// bugsee is not available in debug mode
+  /// * [bugseeToken]: nullable bugsee token, if null bugsee won't be initialized make sure you provide
+  /// [BUGSEE_TOKEN] in the env using `--dart-define` or `launch.json` on vscode
+  Future<void> initialize({
+    String? bugseeToken,
+  });
+
+  /// Manually log a provided exception with a stack trace
+  /// (medium severity exception in Bugsee dashboard)
+  Future<void> logException({
+    required Exception exception,
+    StackTrace? stackTrace,
+  });
+
+  /// Manually log an unhandled exception with a stack trace
+  /// (critical severity exception in Bugsee dashboard)
+  Future<void> logUnhandledException({
+    required Exception exception,
+    StackTrace? stackTrace,
+  });
+
+  /// Manually update the current BugseeEnabled flag in shared prefs and in current manager singleton.
+  Future<void> setIsBugseeEnabled(bool isBugseeEnabled);
+
+  /// Manually update the current enableVideoCapture flag in shared prefs and in current manager singleton.
+  Future<void> setIsVideoCaptureEnabled(bool isBugseeEnabled);
+
+  /// Manually shows the built-in capture log report screen of Bugsee.
+  Future<void> showCaptureLogReport();
+}
+
+final class _BugseeManager implements BugseeManager {
+  final Logger logger;
+  final BugseeRepository bugseeRepository;
+
+  _BugseeManager({
+    required this.logger,
+    required this.bugseeRepository,
+  });
+
+  @override
+  bool isRestartRequired = false;
+
+  @override
+  bool isBugseeEnabled = false;
+
+  @override
+  late bool isVideoCaptureEnabled = false;
+
+  @override
+  bool isConfigurationValid = true;
+
+  late bool _isBugSeeInitialized;
+  BugseeLaunchOptions? launchOptions;
+
+  @override
+  Future<void> initialize({
+    String? bugseeToken,
+  }) async {
+    BugseeConfigurationData bugseeConfigurationData =
+        await bugseeRepository.getBugseeConfiguration();
+
+    launchOptions = initializeLaunchOptions();
+    _isBugSeeInitialized = false;
+
+    if (kDebugMode) {
+      isConfigurationValid = false;
+      logger.i("BUGSEE: deactivated in debug mode");
+      return;
+    }
+
+    if (bugseeToken == null ||
+        !RegExp(bugseeTokenFormat).hasMatch(bugseeToken)) {
+      isConfigurationValid = false;
+      logger.i(
+        "BUGSEE: token is null or invalid, bugsee won't be initialized",
+      );
+      return;
+    }
+
+    if (bugseeConfigurationData.isBugseeEnabled ?? true) {
+      await launchBugseeLogger(bugseeToken);
+    }
+
+    isBugseeEnabled = _isBugSeeInitialized;
+    isVideoCaptureEnabled = _isBugSeeInitialized &&
+        (bugseeConfigurationData.isVideoCaptureEnabled ?? true);
+  }
+
+  Future launchBugseeLogger(String bugseeToken) async {
```

Comment:
Shouldn't this method be private?

Reply:
@jeanplevesque as well for `initializeLaunchOptions` it could be private also